### PR TITLE
[crash-reporter] Actually check if USB developer mode is active. Contribute to JB#37445

### PR DESCRIPTION
**Pull request by martyone**
Note that all but the very last commit belongs to other PRs

### CODE REVIEW

File: src/libs/httpclient/creporternwsessionmgr.cpp [hunk 1]
```diff
@@ -48,8 +48,14 @@ class CReporterNwSessionMgrPrivate
     //! @return Manager of the network configurations provided by the system.
     static QNetworkConfigurationManager &networkManager();
 
+    //! @return USB mode manager
+    static QUsbModed &usbModed();
+
     //! @return True if default connection is in disconnected state.
     static bool connectionIsActive();
```
Comment (pvuorela) [inline src/libs/httpclient/creporternwsessionmgr.cpp]: Wait, what, connection is active when connection is disconnected? Code seems to check activity so comment only wrong?

**Comment by pvuorela** [comment]
I'm a bit puzzled what's the resulting change here, what case is fixed and what is not.

**Comment by martyone** [comment]
As the commit message says "This fixes commit 58e2421 ([HTTP client] re-enable USB networking for upload of the reports)". If you look at that commit you will find that it assumes that even if no connection is active, we MAY be on USB with developer mode on an so it tries sending reports. The problem is the "MAY", so instead of operating silently on background as declared in the documentation "Crash reporter tries to send crash reports before network is ready" (as the bug title says). The change introduced by this PR is to fix the "MAY" for actually checking if the developer mode is active.

**Comment by pvuorela** [comment]
Indeed that earlier commit looks broken. But since network availability is still a guess with developer mode, could have some comment where it's used?

Or, is there even point of having that earlier commit or should it be just reverted? Manually configured usb networking probably breaks this and that in OS, so having one half-working check isn't necessarily worth it?

**Comment by martyone** [comment]
Connman may crash as well.

**Comment by martyone** [comment]
@pvuorela This has been waiting for your reply

**Comment by martyone** [comment]
@pvuorela approved as part of dependent PR #30

**Pull request merged**